src/app/components/doc-viewer/doc-viewer/doc-viewer.component.scss

```diff
@@ -5,7 +5,7 @@ app-doc-viewer{
   right: 0;
   width: 40%;
   background: $theme-bg-color;
-  z-index: 2;
+  z-index: 6;
```

Comment:
Nice. You should make it 7 though, to be on the safe side. I've seen that codemirror sets the vscrollbar and hscrollbar z-index to 6, that's why it overlaps. You should also add that as a comment to that line to explain why we use 7 here instead of 2 🙂